src/mode/modeHandler.ts

```diff
@@ -129,6 +131,40 @@ export class VimSettings {
   useCtrlKeys     = false;
 }
 
+export class ComposeState {
```

Comment:
1. I don't think this should be on VimState, it doesn't really have anything to do with it.
2. This seems like a rather heavyweight wrapper around what is really just a boolean and a string. Just define `class CompositionState { content = ""; isComposing = false; }` and then just use it.